sklego/preprocessing/pandastransformers.py

```diff
@@ -404,40 +435,41 @@ def fit(self, X, y=None):
         Raises
         ------
         TypeError
-            If `X` is not a `pd.DataFrame` object.
+            If `X` is not a supported DataFrame
         KeyError
             If one or more of the columns provided doesn't exist in the input DataFrame.
         ValueError
-            If dropping the specified columns would result in an empty output DataFrame.
+            If provided list of columns to select is empty and would result in an empty output DataFrame.
         """
         self.columns_ = as_list(self.columns)
-        self._check_X_for_type(X)
-        self._check_column_length()
+        X = nw.from_native(X)
         self._check_column_names(X)
+        self.feature_names_ = [x for x in X.columns if x in self.columns_]
```

Comment:
Why is this line needed?

Reply:
I think you don't need `self.feature_names_` at all, it's not in the current code

Reply:
indeed it's not in Attributes list, I see now that the fit performs only checks, cool thank you